### PR TITLE
stop managing global chats subscription if disabled

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Honour the existing "Sync direct and group messages" setting, skipping the attempt to register a global chats subscription accordingly.

#### Ticket Link
None.

### CODE REVIEW

File: server/monitor/subscriptions.go [hunk 1]
```diff
@@ -110,13 +110,33 @@ func (m *Monitor) checkChannelsSubscriptions(msteamsSubscriptionsMap map[string]
...
 		m.api.LogWarn("Unable to get the chat subscriptions from store", "error", err.Error())
 		return
 	}
 
+	// Clean up if we're not syncing direct messages.
+	if !m.syncDirectMessages {
+		// Delete any MS Teams subscription for the global chats, if present.
+		if allChatsSubscription != nil {
+			if err := m.client.DeleteSubscription(allChatsSubscription.ID); err != nil {
+				m.api.LogWarn("Failed to delete old global chats subscriptions", "error", err.Error())
```
Comment (enahum) [inline server/monitor/subscriptions.go:125]: Other than logging, should we do something else like attempting to remove the subscription at a later time?

File: server/monitor/subscriptions.go [hunk 2]
```diff
@@ -110,13 +110,33 @@ func (m *Monitor) checkChannelsSubscriptions(msteamsSubscriptionsMap map[string]
...
+			}
+		}
+
+		// Delete any global subscriptions (assume at most one here).
+		if len(subscriptions) > 0 {
+			if err := m.store.DeleteSubscription(subscriptions[0].SubscriptionID); err != nil {
+				m.api.LogWarn("Unable to delete the old all chats subscription", "error", err.Error())
```
Comment (enahum) [inline server/monitor/subscriptions.go:132]: Same here

**Review by enahum — APPROVED** [review]

File: server/monitor/subscriptions.go (hunk 1 above)
Comment (jespino) [reply server/monitor/subscriptions.go:125]: If this fails, in theory should try it again in the next monitor iteration, It shouldn't be a big deal.

File: server/monitor/subscriptions.go (hunk 2 above)
Comment (jespino) [reply server/monitor/subscriptions.go:132]: Same here, it should try again in the next monitor iteration (in 1 minute or so)

**Review by jespino — APPROVED** [review]
LGTM

**Pull request merged**